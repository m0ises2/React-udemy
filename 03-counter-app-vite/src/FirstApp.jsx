import PropTypes from "prop-types";

export const FirstApp = ( { title, subtitle, age } ) => {
    return (
        <>
            <h1 data-testid='test title'> { title } </h1>
            <p data-testid='test subtitle'> { subtitle } </p>
            <p> { age } years old </p>
        </>
    ) // Esto es un fragmento
}

FirstApp.propTypes = {
    age: PropTypes.number.isRequired,
    subtitle: PropTypes.string.isRequired,
    title: PropTypes.string.isRequired
};

FirstApp.defaultProps = {
    age: 35,
    subtitle: 'Subtitulo por defecto'
};
