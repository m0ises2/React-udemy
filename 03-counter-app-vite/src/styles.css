html, body {
    background-color: rgb(0, 0, 0);
    color: white;
    font-family: Arial, Helvetica, sans-serif;
    font-size: 1.3rem;
    padding: 70px;
}

button {
    padding: 15px;
    font-size: 1.3rem;
}