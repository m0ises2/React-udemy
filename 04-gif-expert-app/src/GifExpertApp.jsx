import { useState } from 'react';
import { AddCategory, GifGrid } from './components';

export const GifExpertApp = () => {
  const [categories, setCategories] = useState([]);
  
  const onAddCategory = (newCategory) => {
    if (categories.includes(newCategory.toLowerCase())) return;
    
    setCategories([newCategory.toLowerCase(),...categories]);
  };

  return (
    <>
      {/* Titulo */}
      <h1>GitExpertApp</h1>

      {/* Input */}
      <AddCategory 
        onNewCategory={ onAddCategory }
      />
      
      {/* Lista de Gif */}
      { 
        categories.map(category => (
          <GifGrid 
            key={ category }
            category={ category }
          />
        ))
      }
        {/* Gif Item */}

    </>
  );
};
