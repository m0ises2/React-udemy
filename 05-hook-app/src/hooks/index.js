export * from './UseCounter';
export * from './useFetch';
export * from './useForm';