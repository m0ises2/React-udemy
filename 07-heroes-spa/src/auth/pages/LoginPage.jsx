import { useContext } from 'react';
import { Navigate, useNavigate } from 'react-router-dom';
import { AuthContext } from '../context'

export const LoginPage = () => {
  const navigate = useNavigate();
  
  const { authState: { logged }, login } = useContext(AuthContext);

  const onLogin = () => {
    login('Mois√©s');

    navigate('/', {
      replace: true
    })
  };

  return (
    logged ?
      <Navigate to='/'/>
    :
    <>
      <div className='container mt-5'>
        <h1> Login </h1>
        <hr />

        <button className='btn btn-primary' onClick={ onLogin }> Login </button>
      </div>
    </>
  )
}
