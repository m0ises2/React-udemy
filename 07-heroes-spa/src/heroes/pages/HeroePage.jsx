
export const HeroePage = () => {
  return (
    <h1>Heroe</h1>
  )
}
