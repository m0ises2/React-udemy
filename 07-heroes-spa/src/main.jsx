import React from 'react';
import ReactDOM from 'react-dom/client';
import './styles.css';
import { AppRouter } from './router/AppRouter';
import { AuthProvider } from './auth';

ReactDOM.createRoot(document.getElementById('root')).render(
  <AuthProvider>
    
    <AppRouter />

  </AuthProvider>,
)
