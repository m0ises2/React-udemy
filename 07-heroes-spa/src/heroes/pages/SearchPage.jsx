import { useLocation, useNavigate } from 'react-router-dom';
import queryString from 'query-string';
import { HeroCard } from '../components';
import { useForm } from '../../hooks/useForm';
import { getHeroesByName } from '../helpers/getHeroesByName';

export const SearchPage = () => {
  const navigate = useNavigate();

  const location = useLocation();

  const { q: heroToQuery = '' } = queryString.parse(location.search);

  const { searchText, onInputChange } = useForm({
    searchText: heroToQuery
  });

  const heroesResults = getHeroesByName(heroToQuery);

  const onSearchSubmit = (event) => {
    event.preventDefault();

    navigate(`?q=${searchText.toLowerCase().trim()}`);
  }

  const showSearch = (heroToQuery.length === 0);
  const showError = (heroToQuery.length > 0) && (heroesResults.length === 0);

  return (
    <>
      <h1> Search </h1>
      <hr />

      <div className='row'>
        <div className='col-5'>
          <h4> Searching </h4>
          <hr />
          <form onSubmit={ onSearchSubmit }>
            <input
              type='text'
              placeholder='Search a Hero'
              className='form-control'
              name='searchText'
              autoComplete='off'
              value={ searchText }
              onChange={ onInputChange }
            />

            <button className='btn btn-outline-primary mt-1'> Search </button>
          </form>
        </div>

        <div className='col-7'>
          <h4> Results </h4>
          <hr />
          {
            // (heroToQuery === '') 
            //   ? <div className='alert alert-primary'> Search a Hero </div>
            //   : (heroesResults.length === 0 ) && <div className='alert alert-danger'> There're not results with { heroToQuery } </div>
          }

          {/* <div className='alert alert-primary' style={{ display: heroToQuery !== '' ? 'none' : '' }}> Search a Hero </div>
          <div className='alert alert-danger' style={{ display: heroesResults.length === 0 ? 'none' : '' }}> There're not results with { heroToQuery } </div> */}

          <div className='alert alert-primary animate__animated animate__fadeIn animate__fast' style={{ display: showSearch ? '' : 'none' }}>
            Search a Hero
          </div>

          <div className='alert alert-danger animate__animated animate__fadeIn animate__fast' style={{ display: showError ? '' : 'none' }}>
            There're not results with { heroToQuery }
          </div>

          {
            heroesResults.map( hero => ( <HeroCard key = {hero.id} {...hero}/> ))
          }
        </div>
      </div>
    </>
  )
}
