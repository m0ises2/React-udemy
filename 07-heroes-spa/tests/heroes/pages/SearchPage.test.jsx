import { render, screen } from '@testing-library/react';
import { SearchPage } from "../../../src/heroes/pages/SearchPage";
import { RouterProvider, createMemoryRouter } from 'react-router-dom';
import { AuthContext } from '../../../src/auth';

describe('SearchPage tests', () => {
  const routes = [
    {
        path: '/',
        element: <SearchPage />,
    }
  ];

  const contextValue = {
    logout: jest.fn(),
    authState: {
        logged: true,
        user: {
          id: 'ABC',
          name: 'Test User'
        }
    }
  };

  test('should match the defaults values agains snapshot', () => {
    const router = createMemoryRouter(routes, {
      initialEntries: ['/']
    });

    const { container } = render(
        <AuthContext.Provider value={ contextValue }>
          <RouterProvider router={router} />
        </AuthContext.Provider>    
    );

    expect(container).toMatchSnapshot();
  });
});
