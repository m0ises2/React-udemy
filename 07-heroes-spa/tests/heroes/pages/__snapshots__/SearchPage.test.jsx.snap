// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchPage tests should match the defaults values agains snapshot 1`] = `
<div>
  <h1>
     Search 
  </h1>
  <hr />
  <div
    class="row"
  >
    <div
      class="col-5"
    >
      <h4>
         Searching 
      </h4>
      <hr />
      <form
        aria-label="form"
      >
        <input
          autocomplete="off"
          class="form-control"
          name="searchText"
          placeholder="Search a Hero"
          type="text"
          value=""
        />
        <button
          class="btn btn-outline-primary mt-1"
        >
           Search 
        </button>
      </form>
    </div>
    <div
      class="col-7"
    >
      <h4>
         Results 
      </h4>
      <hr />
      <div
        aria-label="search-a-hero-msg"
        class="alert alert-primary animate__animated animate__fadeIn animate__fast"
      >
        Search a Hero
      </div>
      <div
        aria-label="no-hero-error-msg"
        class="alert alert-danger animate__animated animate__fadeIn animate__fast"
        style="display: none;"
      >
        There're not results with 
      </div>
    </div>
  </div>
</div>
`;
